hollow/src/main/java/com/netflix/hollow/api/producer/validation/DuplicateDataDetectionValidator.java

```diff
@@ -60,16 +65,47 @@
     private final String[] fieldPathNames;
 
     /**
-     * @param dataTypeName for which this duplicate data detection is needed.
+     * Creates a validator to detect duplicate records of the type that corresponds to the given data type class
+     * annotated with {@link HollowPrimaryKey}.
+     *
+     * @param dataType the data type class
+     * @throws IllegalArgumentException if the data type class is not annotated with {@link HollowPrimaryKey}
+     */
+    public DuplicateDataDetectionValidator(Class<?> dataType) {
+        Objects.requireNonNull(dataType);
+
+        if (!dataType.isAnnotationPresent(HollowPrimaryKey.class)) {
+            throw new IllegalArgumentException("The data class " +
+                    dataType.getName() +
+                    " must be annotated with @HollowPrimaryKey");
+        }
+
+        this.dataTypeName = HollowTypeMapper.getDefaultTypeName(dataType);
+        this.fieldPathNames = null;
+    }
+
+    /**
+     * Creates a validator to detect duplicate records of the type that corresponds to the given data type name.
+     * <p>
+     * The validator will fail, when {@link #onValidate validating}, if a schema with a primary key definition does not
+     * exists for the given data type name.
```

Comment:
typo: I think you mean `exist` (same typo below)

Reply:
Yes! fixed.